### PR TITLE
Adress issues raised by staticcheck and add it to workflow

### PR DESCRIPTION
**Pull request by Jacalz**
Staticcheck is a wonderful tool for performing static analysis of Go programs. This PR addresses all the issues (3 to be exact) that it mentioned and also adds it to the workflow for further checks in the future. I also updated the checkout action to v2 which most importantly has some performance work :)

### CODE REVIEW

**Comment by Jacalz** [comment]
I also added [gosec](https://github.com/securego/gosec) for finding security issues in the code (there are none currently fyi) :)

**Comment by lane-c-wagner** [comment]
Love this contribution. Thank you. Just a couple comments before we merge

File: validate.go [hunk 1]
```diff
@@ -56,10 +56,10 @@ func Validate(password string, minEntropy float64) error {
 
 	if len(allMessages) > 0 {
 		return fmt.Errorf(
-			"Insecure password. Try %v or using a longer password",
+			"insecure password. Try %v or using a longer password",
```
Comment (lane-c-wagner) [inline validate.go:59]: For all of these, it seems weird to capitalize the next sentence be not have the first one capitalized. Can we change it to something like `insecure password, try %v or using a longer password`

Comment (Jacalz) [reply validate.go]: Sorry for a late reply. Haven't had the time to address this until now. Should be fixed now :)

**Pull request merged**